test/support/src/helpers.cc

```diff
@@ -1130,8 +1130,7 @@ void write_array(
   rc = tiledb_query_alloc(ctx, array, TILEDB_WRITE, &query);
   CHECK(rc == TILEDB_OK);
   if (subarray != nullptr) {
-    rc = tiledb_query_set_subarray(ctx, query, subarray);
-    CHECK(rc == TILEDB_OK);
+    query->query_->set_subarray(subarray);
```

Comment:
Since the rest of the code is sitting on top of the C API, so does the rewrite. The replacement function is `tiledb_query_set_subarray_t`

Reply:
Ok, just fixed it!